src/main/kotlin/blackjack/domain/deck/Deck.kt

```diff
@@ -24,7 +24,7 @@ class Deck(private val deckShuffleStrategy: DeckShuffleStarategy) {
     }
 
     private fun shuffleDeck(deck: MutableList<Card>) {
-        deckShuffleStrategy.shuffleDeck(deck)
+        this.deck = deckShuffleStrategy.shuffleDeck(deck).toMutableList()
```

Comment:
방어적 복사를 잘 활용해주셨네요. 👍 
지금도 좋지만, 가급적이면 새로운 컬렉션을 리턴하도록 개선해보면 어떨까요?
```suggestion
        return deckShuffleStrategy.shuffleDeck(deck).toList()
```